configs/.zshrc

```diff
@@ -80,15 +80,34 @@ function update-astroarch()
     # Store actual version
     OLD_VER=$(cat /home/$USER/.astroarch.version)
 
+    # Store the current commit hash before the pull
+    cd /home/$USER/.astroarch
+    CURRENT_COMMIT=$(git rev-parse HEAD)
+
     # Checkout latest changes from git
     cd /home/$USER/.astroarch
     git pull origin main
     cd - > /dev/null 2>&1
 
     NEW_VER=$(cat /home/$USER/.astroarch/configs/.astroarch.version)
 
-    if [ $OLD_VER != $NEW_VER ]; then
-	zsh /home/$USER/.astroarch/scripts/$NEW_VER.sh
+    if [[ "$OLD_VER" != "$NEW_VER" ]]; then
+        # Find all patch scripts, sort them by version, and filter out older versions
+        find /home/$USER/.astroarch/scripts/ -type f -regex '.*/[0-9.]+\.sh' | sort -V | awk -v ver="$OLD_VER" '$0 ~ ver {p=1; next} p' | while read -r SCRIPT; do
```

Comment:
why this change specifically? The output should be exactly the same as the old code? What old versions we are filtering out now?